### DISCUSSION TITLE
How to update ACL to enable sub $SYS/# via Docker / Docker-Compose

### DISCUSSION THREAD

**Discussion by AaronAutomated** [emqx/emqx#7068] [Q&A] [answered]
I've been having some difficulty trying to figure out how to enable a specified user to subscribe to $SYS/# using Docker or Docker-Compose. Could anyone point me in the right direction to achieve this?

**Comment by thalesmg** [comment, marked as answer]
Hi @AaronAutomation !

By default, only clients on `localhost` are allowed to subscribe to `$SYS` topics.

For reference: https://www.emqx.io/docs/en/v4.4/advanced/system-topic.html#sys-system-topic

**Reply by AaronAutomated** [reply]
Thank-you - I referenced this document before. I'd like to change this so that specific users can sub to $SYS topics. I have been unsuccessful trying to achieve this through Docker. Is this not possible?

**Reply by thalesmg** [reply]
Since you are running in Docker, and assuming you are using 4.3.11 or 4.4.0, the default deny rule is probably coming from the [ACL File](https://www.emqx.io/docs/en/v4.4/advanced/acl-file.html#acl-file), which is located by default in `/opt/emqx/etc/acl.conf`.

Simply changing the following line should allow clients to subscribe to `$SYS` (although, as you might have seen in the previous doc, it's not recommended :wink: ):

```diff
-{deny, all, subscribe, ["$SYS/#", {eq, "#"}]}.
+{allow, all, subscribe, ["$SYS/#", {eq, "#"}]}.
``

**Reply by AaronAutomated** [reply]
TY - so there isn't a way to update this through environment variables upon launching an EMQX container?

**Reply by thalesmg** [reply]
Via environment variable, I'm afraid not.

You can mount your custom `acl.conf` when starting the container, though:

```sh
docker run --rm -it -p 8081:8081 -p 18083:18083 -p 1883:1883 -v /tmp/acl.conf:/opt/emqx/etc/acl.conf emqx/emqx:4.3.11
```

**Reply by AaronAutomated** [reply]
Thanks for the help!
